Move WebCore::RunLoopObserver from WebCore/platform/cf to WebCore/platform

whsieh opened the pull request:
#### a1236fc76789874df4b2e8bad5af650024ff385e
<pre>
Move WebCore::RunLoopObserver from WebCore/platform/cf to WebCore/platform
<a href="https://bugs.webkit.org/show_bug.cgi?id=257650">https://bugs.webkit.org/show_bug.cgi?id=257650</a>

Reviewed by Chris Dumez.

Refactor the `RunLoopObserver` helper class such that it&apos;s not inherently limited to ports that use
CoreFoundation; this allows us to deploy it in (otherwise) platform-agnostic code in subsequent
patches, and also opens the door for ports that use different event/run-loop architectures to share
a common helper for observing runloop conditions.

No change in behavior.

* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:

Move the existing logic in `RunLoopObserver.cpp` to `RunLoopObserverCF.cpp`, and reintroduce a
`RunLoopObserver.cpp` to contain platform-agnostic logic in this helper.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/inspector/agents/InspectorTimelineAgent.cpp:
(WebCore::InspectorTimelineAgent::internalStart):
* Source/WebCore/platform/RunLoopObserver.cpp: Added.
(WebCore::RunLoopObserver::~RunLoopObserver):
(WebCore::RunLoopObserver::runLoopObserverFired):
(WebCore::RunLoopObserver::schedule):
(WebCore::RunLoopObserver::invalidate):
* Source/WebCore/platform/RunLoopObserver.h: Renamed from Source/WebCore/platform/cf/RunLoopObserver.h.

Refactor this header file, such that it&apos;s not strictly tied to having CoreFoundation. To ensure
this, we add platform-specific forward declarations and additionally hide the CF-specific behaviors
behind two new WebCore `enum class` types, which then map to `CFIndex` and `CFRunLoopActivity` in
CF-specific code.

(WebCore::RunLoopObserver::RunLoopObserver):
(WebCore::RunLoopObserver::isScheduled const):
* Source/WebCore/platform/cf/RunLoopObserverCF.cpp: Renamed from Source/WebCore/platform/cf/RunLoopObserver.cpp.
(WebCore::cfRunLoopOrder):
(WebCore::cfRunLoopActivity):

Convert from the `RunLoopObserver` WebCore flags to CF-specific values.

(WebCore::RunLoopObserver::runLoopObserverFired):
(WebCore::RunLoopObserver::schedule):
(WebCore::RunLoopObserver::invalidate):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::m_limitsNavigationsToAppBoundDomains):
* Source/WebKit/WebProcess/WebPage/mac/DisplayRefreshMonitorMac.cpp:
(WebKit::DisplayRefreshMonitorMac::startNotificationMechanism):

Currently, this code passes in `kCFRunLoopEntry`, with value 1, as a well-known order (CFIndex).
Since this patch is only intended to refactor existing code, I&apos;m adding a new well-known order for
`DisplayRefreshMonitor` that&apos;s equal to 1, and also adding a FIXME here to track validating whether
or not this is correct behavior.

* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::TiledCoreAnimationDrawingArea):
* Source/WebKitLegacy/mac/WebView/WebViewRenderingUpdateScheduler.mm:
(WebViewRenderingUpdateScheduler::WebViewRenderingUpdateScheduler):

Canonical link: <a href="https://commits.webkit.org/265089@main">https://commits.webkit.org/265089@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6733a8f94f565cd4c0321d37f9db757086ee6e39

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8795 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9083 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9300 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10448 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8783 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/8803 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11070 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9053 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11642 "1 flakes 98 failures") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8941 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9946 "1 flakes 2 failures") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7803 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/10606 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7243 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8046 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/15545 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/8353 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8195 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11523 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/8675 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7075 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7936 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2346 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12148 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8422 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/platform/cf/RunLoopObserverCF.cpp:
```diff
@@ -23,28 +23,56 @@
+static constexpr CFIndex cfRunLoopOrder(RunLoopObserver::WellKnownOrder order)
 {
-    invalidate();
+    constexpr auto coreAnimationCommit = 2000000;
```
I'm guessing this is an invariant... is it defined anywhere?

whsieh replied:
It's an intentionally chosen constant, that matches the runloop order in QuartzCore (it's hard-coded there, and not exposed as any symbol).

cdumez reviewed (approved):
LGTM

whsieh committed:
Move WebCore::RunLoopObserver from WebCore/platform/cf to WebCore/platform
https://bugs.webkit.org/show_bug.cgi?id=257650

Reviewed by Chris Dumez.

Refactor the `RunLoopObserver` helper class such that it's not inherently limited to ports that use
CoreFoundation; this allows us to deploy it in (otherwise) platform-agnostic code in subsequent
patches, and also opens the door for ports that use different event/run-loop architectures to share
a common helper for observing runloop conditions.

No change in behavior.

* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:

Move the existing logic in `RunLoopObserver.cpp` to `RunLoopObserverCF.cpp`, and reintroduce a
`RunLoopObserver.cpp` to contain platform-agnostic logic in this helper.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/inspector/agents/InspectorTimelineAgent.cpp:
(WebCore::InspectorTimelineAgent::internalStart):
* Source/WebCore/platform/RunLoopObserver.cpp: Added.
(WebCore::RunLoopObserver::~RunLoopObserver):
(WebCore::RunLoopObserver::runLoopObserverFired):
(WebCore::RunLoopObserver::schedule):
(WebCore::RunLoopObserver::invalidate):
* Source/WebCore/platform/RunLoopObserver.h: Renamed from Source/WebCore/platform/cf/RunLoopObserver.h.

Refactor this header file, such that it's not strictly tied to having CoreFoundation. To ensure
this, we add platform-specific forward declarations and additionally hide the CF-specific behaviors
behind two new WebCore `enum class` types, which then map to `CFIndex` and `CFRunLoopActivity` in
CF-specific code.

(WebCore::RunLoopObserver::RunLoopObserver):
(WebCore::RunLoopObserver::isScheduled const):
* Source/WebCore/platform/cf/RunLoopObserverCF.cpp: Renamed from Source/WebCore/platform/cf/RunLoopObserver.cpp.
(WebCore::cfRunLoopOrder):
(WebCore::cfRunLoopActivity):

Convert from the `RunLoopObserver` WebCore flags to CF-specific values.

(WebCore::RunLoopObserver::runLoopObserverFired):
(WebCore::RunLoopObserver::schedule):
(WebCore::RunLoopObserver::invalidate):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::m_limitsNavigationsToAppBoundDomains):
* Source/WebKit/WebProcess/WebPage/mac/DisplayRefreshMonitorMac.cpp:
(WebKit::DisplayRefreshMonitorMac::startNotificationMechanism):

Currently, this code passes in `kCFRunLoopEntry`, with value 1, as a well-known order (CFIndex).
Since this patch is only intended to refactor existing code, I'm adding a new well-known order for
`DisplayRefreshMonitor` that's equal to 1, and also adding a FIXME here to track validating whether
or not this is correct behavior.

* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::TiledCoreAnimationDrawingArea):
* Source/WebKitLegacy/mac/WebView/WebViewRenderingUpdateScheduler.mm:
(WebViewRenderingUpdateScheduler::WebViewRenderingUpdateScheduler):

Canonical link: https://commits.webkit.org/265089@main
